packages/binding-netconf/src/async-node-netconf.ts

```diff
@@ -65,63 +65,58 @@ export class Client {
         });
     }
 
-    openRouter() {
-        const self = this;
+    openRouter(): Promise<void> {
         return new Promise((resolve, reject) => {
-            if (self.router.connected) {
+            if (this.router.connected) {
                 // close the old one
                 this.closeRouter();
             }
-            self.router = new nodeNetconf.Client(this.router);
-            self.router.open(function afterOpen(err: string) {
+            this.router = new nodeNetconf.Client(this.router);
+            this.router.open((err?: string) => {
                 if (err) {
                     reject(err);
                 } else {
                     console.debug(
                         "[binding-netconf]",
-                        `New NetConf router opened connection with host ${self.router.host}, port ${self.router.port}, username ${self.router.username}`
+                        `New NetConf router opened connection with host ${this.router.host}, port ${this.router.port}, username ${this.router.username}`
                     );
                     resolve(undefined);
                 }
             });
         });
     }
 
-    rpc(xpath_query: string, method: string, NSs: any, target: string, payload?: any) {
-        const self = this;
+    rpc(xpathQuery: string, method: string, NSs: any, target: string, payload?: any): any {
         return new Promise((resolve, reject) => {
             if (payload) {
-                xpath_query = xpath2json.addLeaves(xpath_query, payload);
+                xpathQuery = xpath2json.addLeaves(xpathQuery, payload);
             }
-            const obj_request = xpath2json.xpath2json(xpath_query, NSs);
-            let final_request: any = {};
-            final_request = JSON.parse(JSON.stringify(METHOD_OBJ[method])); // clone the METHOD_OBJ
+            const objRequest = xpath2json.xpath2json(xpathQuery, NSs);
+            let finalRequest: any = {};
```

Comment:
I know that it might get a little verbose but for JSON parsed objects `unknown` it is a better type than `any`

Reply:
Hmm, I think this does not work with the current code as changing the type to `unknown` leads (for instance) to this error:

```
Property 'edit-config' does not exist on type '{}'.ts(7053)
```